user_guide.adoc

```diff
@@ -1090,7 +1090,117 @@ bytes sparsely.
 
 === Streams and substreams
 
-TODO
+====Introduction and simple example====
+
+A stream is a flow of data from an input file into a parser which is
+generated by a KS script. The parser can request one or more bits of
+data from the stream at a time, but cannot request the same data twice
+and cannot request data be provided out of sequential order. A stream
+knows the maximum amount of data available to be requested by the
+parser and the actual amount of data which has already been
+requested by the parser.
```

Comment:
I'll think of another way to explain streams then, especially with reference to how `pos:` works (seeking) and how `io:` can be used to designate which stream to use.